### PR TITLE
New Timespan object to encapsulate average durations.

### PR DESCRIPTION
**Pull request by pagbabian-splunk**
#### Description of changes:
Added a new object that encapsulates the `duration_avg_xx` attributes with a `type_id` discriminator and a `just_one` constraint.

### CODE REVIEW

**Review by jasonbreimer — DISMISSED** [review]
Looks great! Love the type and type_id addition.

File: objects/timespan_avg.json [hunk 1]
```diff
@@ -0,0 +1,94 @@
+{
+    "caption": "Average Time Span",
+    "description": "The Time Span object represents different average time period durations.",
+    "extends": "object",
+    "name": "timespan_avg",
+    "attributes": {
+        "duration_avg_days": {
```
Comment (Aniak5) [inline objects/timespan_avg.json:7]: Do these need to be added to the dictionary as well? Also, would it make sense to drop `duration_` and just use `*_avg_*` within the dictionary with a generic `See specific usage` description and then specify that we are referring to `duration` within the objects description?

Comment (pagbabian-splunk) [reply objects/timespan_avg.json:7]: I like that - I didn't change any of the dictionary attributes here, Jason created those attributes after some discussion on naming, but as part of an object, we still need an attribute of `timespan` type anyway.  I'm not sure whether the attribute name should also include `x_duration_y` which exists as a scalar (e.g. we have `avg_x` as members of the object, so we don't want that in attribute name).

**Review by Aniak5 — DISMISSED** [review]
LGTM

**Comment by pagbabian-splunk** [comment]
Needed to fix (yet another!!) merge conflict with CHANGELOG.md.

**Review by Aniak5 — APPROVED** [review]

**Review by floydtree — APPROVED** [review]

**Review by mikeradka — APPROVED** [review]

**Pull request merged**